Vega Nullchain breaks on segfault when interacting with snapshot code

When attempting to start up a Vega Nullchain process following recent changes the server now throws a segfault:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102aca74c]

goroutine 172 [running]:
github.com/cosmos/iavl.(*MutableTree).set(0x0, {0x14000055e00, 0xe, 0x10}, {0x1400b65c560, 0xc, 0xc})
        /Users/tom/go/pkg/mod/github.com/cosmos/iavl@v0.19.1/mutable_tree.go:216 +0x4c
github.com/cosmos/iavl.(*MutableTree).Set(0x1041b46a0?, {0x14000055e00?, 0xe?, 0x14000616ac0?}, {0x1400b65c560?, 0x1?, 0x1400b872808?})
        /Users/tom/go/pkg/mod/github.com/cosmos/iavl@v0.19.1/mutable_tree.go:131 +0x28
code.vegaprotocol.io/vega/core/snapshot.(*Engine).snapshotNow(0x14000f69680, {0x104594aa8, 0x1400062f050}, 0x1)
        /Users/tom/Code/vega-market-sim/extern/vega/core/snapshot/engine.go:771 +0x1108
code.vegaprotocol.io/vega/core/snapshot.(*Engine).Snapshot(0x14000f04990?, {0x104594aa8?, 0x1400062f050?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/snapshot/engine.go:664 +0x64
code.vegaprotocol.io/vega/core/processor.(*App).OnCommit(0x14001012dc0)
        /Users/tom/Code/vega-market-sim/extern/vega/core/processor/abci.go:713 +0x228
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).Commit(0x0?)
        /Users/tom/Code/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:76 +0x28
code.vegaprotocol.io/vega/cmd/vega/node.(*appW).Commit(0x14000e3fe20)
        /Users/tom/Code/vega-market-sim/extern/vega/cmd/vega/node/app_wrapper.go:49 +0x30
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x140000d46c0)
        /Users/tom/Code/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:142 +0x108
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleTransaction(0x140000d46c0, {0x1400b836280, 0x254, 0x254})
        /Users/tom/Code/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:158 +0x21c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).SendTransactionSync(0x104594a70?, {0x1400b8076e0?, 0x6?}, {0x1400b836280?, 0x10608d3c8?, 0x70?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:342 +0x2c
code.vegaprotocol.io/vega/core/blockchain.(*Client).SubmitTransactionSync(0x14000fc1a70, {0x104594a70, 0x1400b8076e0}, 0x1400b79a280?)
        /Users/tom/Code/vega-market-sim/extern/vega/core/blockchain/client.go:138 +0x110
code.vegaprotocol.io/vega/core/nodewallets.(*Commander).command.func1()
        /Users/tom/Code/vega-market-sim/extern/vega/core/nodewallets/commander.go:118 +0x5d8
github.com/cenkalti/backoff.RetryNotify(0x1400b79a370, {0x104585e78, 0x1400b791020}, 0x0)
        /Users/tom/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37 +0x1c0
github.com/cenkalti/backoff.Retry(0x0?, {0x104585e78?, 0x1400b791020?})
        /Users/tom/go/pkg/mod/github.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24 +0x28
created by code.vegaprotocol.io/vega/core/nodewallets.(*Commander).command
        /Users/tom/Code/vega-market-sim/extern/vega/core/nodewallets/commander.go:138 +0x19c
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -59,6 +59,7 @@
 - [6060](https://github.com/vegaprotocol/vega/issues/6060) - Fix connection results should not be declared as mandatory in GQL schema.
 - [6097](https://github.com/vegaprotocol/vega/issues/6067) - Fix incorrect asset (metric asset) used for checking market proposer eligibility
 - [6099](https://github.com/vegaprotocol/vega/issues/6099) - Allow recurring transfers with the same to and from but with different asset
+- [6131](https://github.com/vegaprotocol/vega/issues/6131) - `nullblockchain` should call Tendermint Info `abci` to match real flow
 - [6119](https://github.com/vegaprotocol/vega/issues/6119) - Correct order in which market event is emitted
 - [5890](https://github.com/vegaprotocol/vega/issues/5890) - Margin breach during amend doesn't cancel order
 

--- a/core/blockchain/nullchain/mocks/application_service_mock.go
+++ b/core/blockchain/nullchain/mocks/application_service_mock.go
@@ -90,6 +90,20 @@ func (mr *MockApplicationServiceMockRecorder) EndBlock(arg0 interface{}) *gomock
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "EndBlock", reflect.TypeOf((*MockApplicationService)(nil).EndBlock), arg0)
 }
 
+// Info mocks base method.
+func (m *MockApplicationService) Info(arg0 types.RequestInfo) types.ResponseInfo {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "Info", arg0)
+	ret0, _ := ret[0].(types.ResponseInfo)
+	return ret0
+}
+
+// Info indicates an expected call of Info.
+func (mr *MockApplicationServiceMockRecorder) Info(arg0 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Info", reflect.TypeOf((*MockApplicationService)(nil).Info), arg0)
+}
+
 // InitChain mocks base method.
 func (m *MockApplicationService) InitChain(arg0 types.RequestInitChain) types.ResponseInitChain {
 	m.ctrl.T.Helper()

--- a/core/blockchain/nullchain/nullchain.go
+++ b/core/blockchain/nullchain/nullchain.go
@@ -52,6 +52,7 @@ type ApplicationService interface {
 	EndBlock(req abci.RequestEndBlock) (resp abci.ResponseEndBlock)
 	Commit() (resp abci.ResponseCommit)
 	DeliverTx(req abci.RequestDeliverTx) (resp abci.ResponseDeliverTx)
+	Info(req abci.RequestInfo) (resp abci.ResponseInfo)
 }
 
 type NullBlockchain struct {
@@ -121,6 +122,9 @@ func (n *NullBlockchain) ReloadConf(cfg blockchain.Config) {
 }
 
 func (n *NullBlockchain) StartChain() error {
+	// call info to keep the flow the same, and allow things like snapshots to initialise
+	n.app.Info(abci.RequestInfo{})
+
 	err := n.InitChain(n.genesisFile)
 	if err != nil {
 		return err

--- a/core/blockchain/nullchain/nullchain_test.go
+++ b/core/blockchain/nullchain/nullchain_test.go
@@ -199,6 +199,7 @@ func getTestNullChain(t *testing.T, txnPerBlock uint64, d time.Duration) *testNu
 	n.SetABCIApp(app)
 	require.NotNil(t, n)
 
+	app.EXPECT().Info(gomock.Any()).Times(1)
 	app.EXPECT().InitChain(gomock.Any()).Times(1)
 	app.EXPECT().BeginBlock(gomock.Any()).Times(1)
 

